pkg/build/oci/oci.go

```diff
@@ -97,6 +108,24 @@ func BuildImageRefFromLayer(imageRef string, layerTarGZ string, outputTarGZ stri
 	}
 
 	log.Printf("output OCI image file to %s", outputTarGZ)
+	return nil
+}
+
+func PublishImageFromLayer(imageRef string, layerTarGZ string, ic types.ImageConfiguration) error {
+	v1Image, err := buildImageFromLayer(imageRef, layerTarGZ, ic)
+	if err != nil {
+		return err
+	}
+
+	imgRef, err := name.ParseReference(imageRef)
+	if err != nil {
+		return errors.Wrap(err, "unable to parse reference")
+	}
+
+	err = remote.Write(imgRef, v1Image, remote.WithAuthFromKeychain(authn.DefaultKeychain))
```

Comment:
If we incorporate the credential helper stuff here, then we can use this with workload identities without needing to set up a bunch of extra stuff.  This is how `krane` is different from `crane`.

Speaking of, we should be able to drop the `krane push` from `config/task.yaml` with this in.  I can test and send a PR.